### PR TITLE
Support read-only Common Sense Media vocabulary

### PR DESCRIPTION
**Pull request by bethanyaconnor**
See discussion in [Jira](https://codedotorg.atlassian.net/browse/PLAT-746). Add the ability to mark vocabulary as definitions from common sense media. When a vocabulary has `common_sense_media` as true, the definition will be uneditable.

https://user-images.githubusercontent.com/46464143/109852240-6d958f00-7c09-11eb-8027-56af6065db11.mov

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/vocabulary.rb [hunk 1]
```diff
@@ -54,13 +68,14 @@ def summarize_for_edit
...
-      lessons: lessons.map(&:id)
+      lessons: lessons.map(&:id),
+      commonSenseMedia: !!common_sense_media
     }
   end
 
   def generate_key
     return if key
-    self.key = word
+    self.key = common_sense_media ? "#{word}_csm" : word
```
Comment (davidsbailey) [inline dashboard/app/models/vocabulary.rb:78]: nice solution!

**Review by davidsbailey — APPROVED** [review]
LGTM!

**Comment by dmcavoy** [comment]
Does this allow you to have two words that are the same in a course version? one for common sense media and one for "normal"?

**Comment by bethanyaconnor** [comment]
> Does this allow you to have two words that are the same in a course version? one for common sense media and one for "normal"?

Yep! We prevent multiple definitions per word by disallowing duplicate keys in a course version. The `_csm` suffix to the key for common sense media definitions allows for two definitions for one word

**Review by dmcavoy — APPROVED** [review]

**Pull request merged**